### ISSUE TITLE
Log but no entries

### ISSUE THREAD

**Issue by vr8hub** [jongpie/NebulaLogger#199]
Sorry, apparently I'm this week's problem child.

I have two "Add Log Entry for SObject Record Collection" actions in a flow, and only those actions. Both have Save Log set to True, both with Logging Level set to DEBUG. Both are in the path that will get called. (Confirmed with debug before I put the logging in.)

The Flow completes successfully. However, the Logger console shows me only a Log__c entry, no Log Entries. After the first time it happened, I deleted the log (so no logs at all) and ran the flow again, same result. Since I'm only calling actions that should create a log entry, I don't know how that could be happening; if the log actions weren't getting hit at all, there wouldn't be a log record, but since there's a log record, that implies that they are getting hit, but there's no log entry records, which the actions should both be creating.

I have the Logger: Admin permission set assigned to my user. (I'm a sysadmin.) I also created a Logger Setting entry for my user (and only that one, no org default), with Logging Level set to DEBUG and Is Enabled checked, everything else defaulted (Read, 14, EVENT_BUS, etc.). With my level set to DEBUG and the flow actions set to DEBUG, my understanding is that they should be created.

What am I missing and/or doing wrong?

**Comment by vr8hub** [comment]
I turned off the logging level on the two actions so that it used the default (which is?), but it didn't make any difference. Still only a log created, no log entries.

**Comment by jongpie** [comment]
Hey @vr8hub - it sounds like you're doing everything correctly,  so it could be a bug of some sort. I've seen similar issues happen before where the `Log__c` record is created, but the `LogEntry__c` records fail - these failures happen asynchronously, so you typically won't see the errors when the Flow runs.

One option you can try is updating your Logger Setting entry to use SYNCHRONOUS_DML as the default save method (instead of EVENT_BUS), and then try running your Flow again - that will (hopefully) show you whatever error is happening that's preventing the `LogEntry__c` records from saving. If that works, let me know what the error is and I can help from there.

**Comment by vr8hub** [comment]
I made the change, but no difference—I still got just a Log record with no LogEntries, and no error. I checked and the SysAdmin profile has full privileges on both objects (Log and LogEntry). I also added a plain "Add Log Entry" to see if it was isolated to the record collection action; it's not, i.e. I still didn't get any log entries even after that (total of three actions, one Add Log, two Add Log Record Collection).

**Comment by vr8hub** [comment]
So sorry, spoke too soon. I didn't notice—something (SF?) created an org default for the Logger settings, and it was that I changed to sync, not my user settings. So I deleted my user settings since SF apparently doesn't like a user having settings when there isn't an org default. I ran it again, and this time got an error:
Error Occurred: An Apex error occurred: System.DmlException: Insert failed. First exception on row 0; first error: CANNOT_INSERT_UPDATE_ACTIVATE_ENTITY, LogEntry: data changed by trigger for field Flow Description: data value too large: (value has been hidden): []

**Comment by jamessimone** [comment]
@vr8hub we can arrive at a more elegant solution for this, but in the meantime, if you navigate to the `LogEntryEventHandler` class, if you update line 215 from:

```java
insert this.logEntries;
```

to:

```java
Database.DMLOptions dmlOptions = new Database.DMLOptions();
dmlOptions.AllowFieldTruncation = true;
 Database.insert(this.logEntries, dmlOptions);
```

That will get you back on track for now.

**Comment by vr8hub** [comment]
So the LogEntryHandler class assigns the FlowDescription in LogEntry to the description from FlowDefinitionView, and in running that query manually, this particular flow's description (and several others of ours) have a description that's longer than 255 characters. So I'm guessing that assignent is what's blowing up.

**Comment by vr8hub** [comment]
Ahh, very good James, I posted before I saw your message. I'll make that change, thanks for your help!

**Comment by jamessimone** [comment]
no problem, happy to help. Truncated field values is something we've looked at in other areas of logger, but in thinking about it, it makes sense that the Flow description could exceed 255 characters - we'll probably need to do an audit for other areas where this error might present itself! 😅

**Comment by vr8hub** [comment]
Before I make that change… is that going to work? If the assignment is what's causing the problem, isn't the insert too late to catch it? IOW, isn't it dying when the assignment is made, rather than when the insert happens?

**Comment by vr8hub** [comment]
I just added .left(255) to the end of the assignment in LogEntryHandler, and that got rid of the error, and lo and behold I have Log Entries. Yea!

Thank you both for all your help, and sorry for all the trouble!

**Comment by jongpie** [comment]
@vr8hub no trouble at all, thanks so much for all of the feedback! I'll include a fix for this issue when I also fix #198.

Let us know if you run into any other problems!
